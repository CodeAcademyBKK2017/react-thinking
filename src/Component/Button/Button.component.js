import React, {Component} from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';

class Button extends Component {
  render () {
    return (
      <button onClick={this.props.func}>{this.props.text}</button>
    );
  }
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  func: PropTypes.func.isRequired
};

Button.defaultProps = {
  text: '',
  func: noop
};

export default Button;