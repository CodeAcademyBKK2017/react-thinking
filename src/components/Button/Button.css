.btn {
    display: inline-block;
    width: 180px;
    height: 100px;

    font-size: 24px;
    font-weight: bold;
    text-align: center;
    line-height: 100px;
    border-radius: 8px;
    border: 3px solid black;
    cursor: pointer;
}
.btn.btn-add {
    float:left;
    background: #ffaf04;
    color: black;
}
.btn.btn-subtract { 
    float:right;
    background: #087e29;
    color: white;
}