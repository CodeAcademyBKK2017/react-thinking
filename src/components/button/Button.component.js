import React, { Component } from 'react';
import './Button.style.css';
import ProptTypes from 'prop-types';
import noop from 'lodash/noop';

class Button extends Component {

 render() {
  const {func,funcName} = this.props
   return (
    <button onClick={func}>{funcName}</button>
   );
 }
}

Button.proptTypes = {
    func: ProptTypes.oneOf(['ADD','SUBTRACT']).isRequired,
    funcName: ProptTypes.func.isRequired
  };
  
Button.defaultProps = {
    funcName: '',
    func: noop
};
export default Button;