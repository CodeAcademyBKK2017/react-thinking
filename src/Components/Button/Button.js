import React, {Component} from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';
import './Button.css';

class Button extends Component {
render () {
    const {onclick,type} = this.props;
    return (
        <div className={`btn btn-${type}`} onClick={onclick}>{this.props.title}</div>
    );
}
}

Button.propTypes = {
    type: PropTypes.oneOf(['add','subtract']).isRequired,
    onclick: PropTypes.func.isRequired
};

Button.defaultProps = {
    onclick: noop
};
export default Button;