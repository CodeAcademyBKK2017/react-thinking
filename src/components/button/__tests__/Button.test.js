import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Button from '../Button.component';

describe('Button component', () => {
  it('Button: renders correctly', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});