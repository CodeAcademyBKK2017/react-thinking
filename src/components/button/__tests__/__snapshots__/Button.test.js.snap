// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button: renders correctly 1`] = `
<button
  onClick={[Function]}
>
  
</button>
`;
